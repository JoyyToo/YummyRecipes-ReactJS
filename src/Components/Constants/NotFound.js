import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
    <div style={{textAlign: 'center'}}>
    <h1>Oops</h1>
    <div>
        <h2 >Page Not Found, 404. Sorry, there is nothing here.</h2>
        <div>
          Visit our <button><Link to="/" >HOMEPAGE</Link></button></div>
          </div>
        </div>
  );

export default NotFound;