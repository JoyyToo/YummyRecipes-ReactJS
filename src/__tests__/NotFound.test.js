import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import NotFound from '../Components/Constants/PagenotFound'

describe('NotFound component', () => {
    const wrapper = shallow(<NotFound />);
  
    it('renders properly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('contains div', () => {
        expect(wrapper.find('div')).toHaveLength(3);
    });

    it('contains h2', () => {
        expect(wrapper.find('h2')).toHaveLength(1);
    });

    it('contains h1', () => {
        expect(wrapper.find('h1')).toHaveLength(1);
    });

    it('contains button', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    });

    it('contains Link', () => {
        expect(wrapper.find('Link')).toHaveLength(1);
    });

})