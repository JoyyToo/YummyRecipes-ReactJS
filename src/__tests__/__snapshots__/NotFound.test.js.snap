// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component renders properly 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Oops
  </h1>
  <div>
    <h2>
      Page Not Found, 404. Sorry, there is nothing here.
    </h2>
    <div>
      Visit our 
      <button>
        <Link
          replace={false}
          to="/"
        >
          HOMEPAGE
        </Link>
      </button>
    </div>
  </div>
</div>
`;
